### PR TITLE
fix(releases): Hide release manifest

### PR DESCRIPTION
**Pull request by jjbayer**
Hide release manifest introduced in https://github.com/getsentry/sentry/pull/26529 from API endpoints and counts.

### CODE REVIEW

**Review by jan-auer — COMMENTED** [review]
I suspect that the join on `File` is going to be too expensive in practice. The workaround would be a (dangerous) migration on `ReleaseFile` to introduce a type column there.

@beezz can you chime in here, since you recently looked into this on `EventAttachment`.

**Comment by jjbayer** [comment]
> I suspect that the join on `File` is going to be too expensive in practice.

@jan-auer it might be worth noting that most use cases of `public_objects()` need the `File` object anyway, either explicitly like [here](https://github.com/getsentry/sentry/pull/26560/files#diff-68f9f3d21eb73b013d0c08d9266fc470e17212b16cb11d25faf8af2d871dc5b8L60), or implicitly in the serializer, like [here](https://github.com/getsentry/sentry/blob/ce898db90cf0a79ef3ec9991683bccdfaf8c693b/src/sentry/api/serializers/models/release_file.py#L12).

**Comment by EvanPurkhiser** [comment]
hi mom

File: src/sentry/models/releasefile.py [hunk 1]
```diff
@@ -101,6 +101,11 @@ def normalize(cls, url):
             urls.append("~" + urlunsplit(uri_relative_without_query))
         return urls
 
+    @classmethod
+    def public_objects(cls):
```
Comment (beezz) [inline src/sentry/models/releasefile.py:105]: nit: I think this is a task for a [Manager](https://docs.djangoproject.com/en/3.2/topics/db/managers/), then it's easier to chain in existing query with filters.
- in sentry e.g. - https://github.com/getsentry/sentry/blob/master/src/sentry/models/project.py#L53

**Review by beezz — APPROVED** [review]

File: src/sentry/models/releasefile.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/models/releasefile.py]: Thanks for the hint, I rewrote it now according to this: https://docs.djangoproject.com/en/1.11/topics/db/managers/#modifying-a-manager-s-initial-queryset

**Pull request merged**